spec/ddtrace/contrib/sidekiq/support/helper.rb

```diff
@@ -25,11 +38,43 @@
 
     Sidekiq::Testing.inline!
   end
+end
 
-  let!(:empty_worker) do
-    stub_const('EmptyWorker', Class.new do
-      include Sidekiq::Worker
-      def perform; end
-    end)
+module SidekiqServerExpectations
+  include SidekiqTestingConfiguration
+
+  def expect_in_sidekiq_server(duration: 2, &expectations)
+    expect_in_fork(
+      # Due to the expectations being run in an `at_exit` hook, to have visibility into the
+      # spans created in the forked process, the exit status will be 0 even if our expectations fail.
+      # Instead, look at `STDERR`, ignoring warnings.
+      fork_expectations: proc do |status:, stdout:, stderr:|
+        expect(status).to be_success, "STDOUT:`#{stdout}` STDERR:`#{stderr}"
+
+        non_warning_testing_stderr = stderr.split("\n").reject { |line| line.include?(': warning:') }
+
+        expect(non_warning_testing_stderr).to be_empty, "STDOUT:`#{stdout}` STDERR:`#{stderr}"
+      end
+    ) do
+      at_exit(&expectations)
+
+      # NB: This is needed because we want to patch within a forked process.
+      if Datadog::Contrib::Sidekiq::Patcher.instance_variable_get(:@patch_only_once)
+        Datadog::Contrib::Sidekiq::Patcher.instance_variable_get(:@patch_only_once).send(:reset_ran_once_state_for_tests)
+      end
+
+      require 'sidekiq/cli'
+
+      configure_sidekiq
+
+      Thread.new do
+        cli = Sidekiq::CLI.instance
+        cli.parse(['--require', 'spec/ddtrace/contrib/sidekiq/support/environment.rb']) # boot the "app"
```

Comment:
If `spec/ddtrace/contrib/sidekiq/support/environment.rb` is always empty, would you be able to remove `spec/ddtrace/contrib/sidekiq/support/environment.rb` from the git repo and instead create a [temporary empty file](https://ruby-doc.org/stdlib-2.4.0/libdoc/tempfile/rdoc/Tempfile.html) around this line?

If Sidekiq will only read it at an unspecified time in the future, then no need to bother with this suggestion.

Reply:
Great idea! I've switched this to a `Tempfile` and removed the empty `environment.rb` file.